xs/src/libslic3r/ConfigBase.hpp

```diff
@@ -317,9 +351,19 @@ class ConfigOptionFloatOrPercent : public ConfigOptionPercent
     ConfigOptionFloatOrPercent() : ConfigOptionPercent(0), percent(false) {};
     ConfigOptionFloatOrPercent(double _value, bool _percent)
         : ConfigOptionPercent(_value), percent(_percent) {};
-    ConfigOptionFloatOrPercent* clone() const { return new ConfigOptionFloatOrPercent(this->value, this->percent); };
+    ConfigOptionFloatOrPercent* clone() const override { return new ConfigOptionFloatOrPercent(this->value, this->percent); };
     
-    void set(const ConfigOption &option) {
+    double getFloat() const override { throw BadOptionTypeException(); };
```

Comment:
Wouldn't this be appropriate to only throw `BadOptionTypeException` if it's not a percent?

Reply:
But you don't know whether it's a percent or not before calling this :)
(You usually call `ConfigBase::getFloat()` which in turn calls this. If you dereference the option via `config->option(opt_key)` you can then call `->value` and `->percent` directly)

Reply:
The option knows the state of its own percent flag, though.